prodavnica/src/main/java/com/levi9/prodavnica/serviceImpl/BookServiceImpl.java

```diff
@@ -130,4 +131,20 @@ public boolean addBook(AddUpdateBookDTO addUpdateBookDTO) {
 		return true;
 	}
 
+	@Override
+	public BookListDTO searchForBook(String search) {
+		BookListDTO bookListDTO = new BookListDTO();
+		List<Book> books = bookRepository.getBookSearch(search);
+		if (!books.isEmpty()) {
+			for (Book book : books)
+				if (!book.isDeleted() && (book.getAmount() > 0) && bookListDTO.getBooks().stream().noneMatch(b -> b.getBookId().equals(book.getBookId())))
+					bookListDTO.getBooks().add(bookMapper.map(book));
+		}else{
+			throw  new StoreException(HttpStatus.NOT_FOUND,"Book doesn't exist!");
```

Comment:
Maybe just return an empty object when search does not find anything, instead of throwing an error. It is more user friendly, since user then needs to handle the error somehow.